### DISCUSSION TITLE
Mender install on Jetson Xavier NX 16GB failed

### DISCUSSION THREAD

**Discussion by carlessilva** [OE4T/tegra-demo-distro#208] [Q&A] [answered]
Hi,
I am testing a custom distro based on tegra-demo-distro (dunfell branch) on Xavier NX 16GB commercial module.
When you try to update the distro using Mender install the process fails.
It's the same error reported on this discussion [#172](https://github.com/OE4T/tegra-demo-distro/discussions/172) related to TX2i module.

The Mender update works fine on Xavier NX 8GB commercial module. 

If I have understood correctly, the problem is related to BOARDSKU 
The 'jetson-xavier-nx-devkit-emmc.conf' file defines TEGRA_BOARDSKU ?= "0001" which is ok for 8GB module
The 16GB module BOARDSKU is "0003" (P3668-0003)
[NVIDIA docs](https://docs.nvidia.com/jetson/archives/r34.1/DeveloperGuide/index.html)

Please any help for solving this issue?
Many thanks in advance

$ sudo mender install mantos-image-ai-jetson-xavier-nx-devkit-emmc.mender 
INFO[0000] Loaded configuration file: /var/lib/mender/mender.conf 
INFO[0000] Loaded configuration file: /etc/mender/mender.conf 
INFO[0000] Mender running on partition: /dev/mmcblk0p1  
INFO[0000] Start updating from local image file: [mantos-image-ai-jetson-xavier-nx-devkit-emmc.mender] 
Installing Artifact of size 1838835712...
INFO[0000] No public key was provided for authenticating the artifact 
INFO[0000] Opening device "/dev/mmcblk0p11" for writing 
INFO[0000] Native sector size of block device /dev/mmcblk0p11 is 512 bytes. Mender will write in chunks of 1048576 bytes 
.............................................................. - 100 %
INFO[0062] All bytes were successfully written to the new partition 
INFO[0062] The optimized block-device writer wrote a total of 4742 frames, where 0 frames did need to be rewritten (i.e., skipped) 
INFO[0063] Wrote 4971769856/4971769856 bytes to the inactive partition 
INFO[0063] Enabling partition with new image installed to be a boot candidate: 11 
INFO[0063] Executing script: ArtifactInstall_Leave_80_bl-update 
INFO[0064] Collected output (stderr) while running script /var/lib/mender/scripts/ArtifactInstall_Leave_80_bl-update
Error: missing entries for partitions: bpmp-fw-dtb, bootloader-dtb, VER, mb1, BCT, MB1_BCT, MEM_BCT, kernel-dtb
       for TNSPEC 3668-301-0003-B.0-1-2-jetson-xavier-nx-devkit-emmc-mmcblk0p1
ERR: bootloader update failed

---------- end of script output 
ERRO[0064] ArtifactInstall_Leave script failed: statescript: error executing 'ArtifactInstall_Leave_80_bl-update': 1 : exit status 1 
Rolling back Artifact...
INFO[0064] Rolling back to the inactive partition (11). 
ERRO[0064] statescript: error executing 'ArtifactInstall_Leave_80_bl-update': 1 : exit status 1

**Comment by madisongh** [comment, marked as answer]
Sounds like the `TEGRA_BUPGEN_SPECS` setting in the machine config file needs updating to include a line for the 16GB board SKU.

**Reply by carlessilva** [reply]
That's correct.
Updating TEGRA_BUPGEN_SPECS for including **fab=301;boardsku=0003;boardrev=** solves the problem.
As a workaround I have set the new value in my own customdistro.conf file and I have checked that mender update works fine.
Maybe you can update jetson-xavier-nx-devkit-emmc.conf file of meta-tegra layer for fixing the problem in future commits.

Thanks a lot Matt!

**Reply by madisongh** [reply]
I've updated the `jetson-xavier-nx-devkit-emmc.conf` file with this addition in the meta-tegra master, kirkstone, and dunfell branches, so next time you update you should be able to drop your workaround.
